util/test/chpl_plot.py

```diff
@@ -0,0 +1,784 @@
+#     |  8.0 |  12.0 |   22.0 |   32.0 |
+#     | 16.0 |  13.0 |   23.0 |   33.0 |
+#     | 32.0 |  14.0 |   24.0 |   34.0 |
+#
+# In such situations you may want to consider relabeling the columns before
+# conducting the joins or you can use the relabel_col_w_tbl_title on the tables
+# before conducting the join.
+#
+#
+# F.A.Q.
+#
+#  (Q) The way my plot is being visualized isn't right. I need to tweak it,
+#      what can I do?
+#  (A) First look through this FAQ to see if another answer can give more
+#      direct help. If not, consider storing the plot object to a variable
+#      without saving it yet like so: `plot = tbl.plot(..., save=False)`
+#      (where ... is the set of arguments you would pass to the `plot`
+#      function otherwise). Once you have this, you can call various
+#      functions on to modify it, and then call plot.save() to save it once
+#      you're done.
+#      
+#      Take a look at the Plot API first to see if it has what you need.
+#      If not you can access the underlying matplotlib objects `plot.ax` or
+#      `plot.fig`.
+#
+#  (Q) While debugging, how can I see the content in my table object? Doing
+#      `print(tbl)` shows me its a table object but doesn't list the data?
+#  (A) Do: `print(tbl.md())`, "md" here means markdown and returns a string
+#      of the table in markdown format.
+#
+#  (Q) Parts of my plot are clipped in my image (labels, etc.). What can I do?
+#  (A) Get the plot object without saving it by calling
+#      `p = plot(...., save=False)`. You can then forward 'tight' to
+#       matplotlib's 'bbobx_unches' argument when saving the plot like so:
+#       `p.save(bbox_inches='tight')`. If the ends up cutting other things
+#       off you can try adding extra paddingl ike this:
+#       `p.save(bbox_inches='tight', pad_inches=0.5)`
+#
+#  (Q) My legend isn't placed very nicely. I want to move it; what can I do?
+#  (A) Separate out plotting and saving (as discussed in the previous question)
+#      and you can set `plot.mpl_legend_args` to a dictionary for the kwargs
+#      that would get passed to matplotlib's 'Axes.legend' function. Keep in
+#      mind that this script will pass `legend` arguments for 'loc', 'fontsize',
+#      and 'ncol', so if you do not want these to be passed you'll need to
+#      overwrite them with 'None'. See here for more info about the aguments
+#      that can be passed to Axes.legend:
+#      https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.legend.html
+#      As an example, the following will give the legend 3 columns and center it
+#      above and outside of the chart:
+#
+#        p.mpl_legend_args = { 'loc': 'center',
+#                              'ncol': 3,
+#                              'bbox_to_anchor': (0.5, 1.3) }
+
+
+
+import os, re, codecs, shlex, copy, numbers, sys
+from functools import reduce
+
+import numpy as np
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import matplotlib.ticker as ticker
```

Comment:
Would be nice to note in some of the docs that matplotlib needs to be installed. Maybe even a requirements file to help with the install?